Project - Marco Boso

MarcoBoso opened the pull request.

MarcoBoso commented:
i'm sending the entire project

Enchufa2 commented on _projects/2024/100535153/100535153.Rmd:
````diff
@@ -0,0 +1,941 @@
+    toc: true
+---
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(out.width="1000%", fig.align="center")
````
100%?

Enchufa2 commented on _projects/2024/100535153/100535153.Rmd:
````diff
@@ -0,0 +1,941 @@
+
+### Load the packages 
+```{r packages}
+#| echo: true
````
Please remove any "echo" option throughout the post. True is the default, and there should be no chunk with false, because we always want to show the code.

Enchufa2 commented on _projects/2024/100535153/100535153.Rmd:
````diff
@@ -0,0 +1,941 @@
+### Load the packages 
+```{r packages}
+#| echo: true
+library(tidyverse)
+library(ggplot2)
+library(dplyr)
+library(ggthemes)
+library(patchwork)
+library(grid)
+library(gridExtra)
+library(openxlsx)
+library(readxl)
+library(cowplot)
+library(readxl)
+library(extrafont)
+library(showtext)
````
Are all these libraries necessary? Just curious. If that's true, it's fine. But I see e.g. openxlsx and readxl, and both are for the same thing.

Enchufa2 commented on _projects/2024/100535153/100535153.Rmd:
````diff
@@ -0,0 +1,941 @@
+```{r, echo = FALSE}
+data1 <- read_excel("C:/Users/marco/OneDrive/Desktop/Data Visualization - I sem/Project - The changing nature of middle-class job/Changing nature of middle class.xlsx")
+
+data2 <- read_excel("C:/Users/marco/OneDrive/Desktop/Data Visualization - I sem/Project - The changing nature of middle-class job/Changing nature of middle class.xlsx", sheet = "Managers and administrators")
+
+data3 <- read_excel("C:/Users/marco/OneDrive/Desktop/Data Visualization - I sem/Project - The changing nature of middle-class job/Changing nature of middle class.xlsx", sheet = "Service workers and salespeople")
+
+data4 <- read_excel("C:/Users/marco/OneDrive/Desktop/Data Visualization - I sem/Project - The changing nature of middle-class job/Changing nature of middle class.xlsx", sheet = "Technicians")
+
+data5 <- read_excel("C:/Users/marco/OneDrive/Desktop/Data Visualization - I sem/Project - The changing nature of middle-class job/Changing nature of middle class.xlsx", sheet = "Police, fire and military")
+
+data6 <- read_excel("C:/Users/marco/OneDrive/Desktop/Data Visualization - I sem/Project - The changing nature of middle-class job/Changing nature of middle class.xlsx", sheet = "Administrative support workers")
+
+data7 <- read_excel("C:/Users/marco/OneDrive/Desktop/Data Visualization - I sem/Project - The changing nature of middle-class job/Changing nature of middle class.xlsx", sheet = "Transportation workers")
+
+data8 <- read_excel("C:/Users/marco/OneDrive/Desktop/Data Visualization - I sem/Project - The changing nature of middle-class job/Changing nature of middle class.xlsx", sheet = "Construction and agricolture ")
+
+data9 <- read_excel("C:/Users/marco/OneDrive/Desktop/Data Visualization - I sem/Project - The changing nature of middle-class job/Changing nature of middle class.xlsx", sheet = "Skilled production workers")
+
+data10 <- read_excel("C:/Users/marco/OneDrive/Desktop/Data Visualization - I sem/Project - The changing nature of middle-class job/Changing nature of middle class.xlsx", sheet = "Machine operators and assembler")
````
Put the datasets close to the Rmd and add them to the pull request. Then don't use paths in your computer, but just the names of the files.

Enchufa2 commented on _projects/2024/100535153/100535153.Rmd:
````diff
@@ -0,0 +1,941 @@
+Here you can find the link to the page of the article to extract the data: [The Changing Nature
+of Middle-Class Jobs - The New York Times ](https://www.nytimes.com/interactive/2015/02/23/business/economy/the-changing-nature-of-middle-class-jobs.html)
+
+```{r, echo = FALSE}
````
Remove echo option.

Enchufa2 commented on _projects/2024/100535153/100535153.Rmd:
````diff
@@ -0,0 +1,941 @@
+This is an example of the sheet for 'Managers and administrators'. The column 'Jobs' represents specific occupations or job titles within each sector, while 'Sector' groups jobs into broader categories and reflects the division by sheets. 'Years' indicates the time frame of the data (1980 and 2012), and 'Employed' shows the number of people employed in a specific job, expressed in thousands (e.g. ...
+
+```{r showing data}
+data1
````
`head(data1)`. A small peek is enough.

Enchufa2 commented on _projects/2024/100535153/100535153.Rmd:
```diff
@@ -0,0 +1,941 @@
+  labs(x = "- 62 jobs\nMachine\noperators and\nassemblers") +
+  
+  # Minimal theme for cleaner visuals
+  theme_minimal() +  
+  theme(
+    panel.grid.major.x = element_blank(),  
+    panel.grid.minor = element_blank(),    
+    panel.grid.major.y = element_blank(), 
+    legend.position = "none",              
+    axis.text.y = element_text(size = 10, color = "black"),
+    axis.text.x = element_blank(),
+    axis.title.y = element_blank(),
+    axis.title = element_text(size = 12)
+  )
```
All the graphs share most of the code. So it would be nicer if you could save the common parts in a variable, and then just reuse it.

MarcoBoso replied:
Thanks for the input! I have created two variables that contain the parts in the codes that were constant and redundant in the 10 graphs. I preferred to keep the first graph divided by ‘steps’ to give at the same time an explanation of all the parts in the code of the graph

Enchufa2 commented on _projects/2024/100535153/100535153.Rmd:
````diff
@@ -0,0 +1,941 @@
+
+Finally, we can add the title and the subtitle to the graph.
+
+```{r replica of the original graph, layout="l-page", fig.height=10.6, fig.width=19.2, preview=TRUE}
````
Chunk names shouldn't have spaces (they're problematic). So it should be `replica-of-the-original-graph` instead. Please apply this to all chunks.

Enchufa2 commented on _projects/2024/100535153/100535153.Rmd:
````diff
@@ -0,0 +1,941 @@
+```
+Even though this chart does not delve into the detailed dynamics of occupational changes within middle-class jobs, I believe it offers a fresh perspective on the issue. By focusing on the overall percentage growth or decline of individual jobs, it highlights the broader shifts in employment patterns over time.
+
+Distill is a publication format for scientific and technical writing, native to the web.
+
+Learn more about using Distill at <https://rstudio.github.io/distill>.
````
Remove this.

Enchufa2 reviewed (changes requested):
Great work! Some comments below.

MarcoBoso commented:
improvements: deleted all echoes, deleted the libraries I did not use in the project, inserted the dataset with the original data and changed it, fixed the names of the chunks and other small changes

Enchufa2 reviewed (approved):
Thanks, mergin after a few adjustments!

Enchufa2 commented:
Here's your work, congrats! https://csslab.uc3m.es/dataviz/projects/2024/100535153/
